kDrive/Utils/MatomoUtils+UI.swift

```diff
@@ -0,0 +1,103 @@
+/*
+ Infomaniak kDrive - iOS App
+ Copyright (C) 2024 Infomaniak Network SA
+
+ This program is free software: you can redistribute it and/or modify
+ it under the terms of the GNU General Public License as published by
+ the Free Software Foundation, either version 3 of the License, or
+ (at your option) any later version.
+
+ This program is distributed in the hope that it will be useful,
+ but WITHOUT ANY WARRANTY; without even the implied warranty of
+ MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ GNU General Public License for more details.
+
+ You should have received a copy of the GNU General Public License
+ along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Foundation
+import kDriveCore
+
+extension MatomoUtils {
+    // MARK: - Share and Rights
+
+    static func trackRightSelection(type: RightsSelectionType, selected right: String) {
+        switch type {
+        case .shareLinkSettings:
+            MatomoUtils.track(eventWithCategory: .shareAndRights, name: "\(right)ShareLink")
+        case .addUserRights, .officeOnly:
+            if right == UserPermission.delete.rawValue {
+                MatomoUtils.track(eventWithCategory: .shareAndRights, name: "deleteUser")
+            } else {
+                MatomoUtils.track(eventWithCategory: .shareAndRights, name: "\(right)Right")
+            }
+        }
+    }
+
+    static func trackShareLinkSettings(protectWithPassword: Bool, downloadFromLink: Bool, expirationDateLink: Bool) {
+        MatomoUtils.track(eventWithCategory: .shareAndRights, name: "protectWithPassword", value: protectWithPassword)
+        MatomoUtils.track(eventWithCategory: .shareAndRights, name: "downloadFromLink", value: downloadFromLink)
+        MatomoUtils.track(eventWithCategory: .shareAndRights, name: "expirationDateLink", value: expirationDateLink)
+    }
+
+    // MARK: - File action
+
+    #if !ISEXTENSION
```

Comment:
Do we still need the ifdef ? I would be great to remove them where we can as Xcode seems a bit lost with the code.

Reply:
If I remove the ifdef, the extensions are not seeing `FloatingPanelAction`. I can dig deeper or stop there.

Reply:
If you need to do a bigger refactor it's not worth it for now